testing/yuca/ca/test_nca.py

```diff
@@ -0,0 +1,174 @@
+import unittest
+
+import numpy as np
+
+import torch
+
+from yuca.ca.neural import NCA
+
+
+class TestNCA(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def test_multiverse_forward(self):
+
+        ca = NCA()
+        ca.no_grad()
+
+        ca.default_init()
+
+        for vectorization in [1,2,4,9]:
+
+            ca.to_device("cpu")
+
+            x = torch.randn(vectorization, 1, 32, 32)
+            new_x = ca.forward(x)
+
+            self.assertEqual(new_x.shape, x.shape)
+            
+            if torch.cuda.is_available():
+
+                ca.to_device("cuda")
+
+                x = torch.randn(vectorization, 1, 32, 32).to("cuda")
+                new_x = ca.forward(x)
+
+                self.assertEqual(new_x.shape, x.shape)
+                
+        ca = NCA()
+        ca.default_init()
+
+        for vectorization in [1,2,4,9]:
+
+            ca.to_device("cpu")
+
+            x = torch.randn(vectorization, 1, 32, 32)
+            new_x = ca.forward(x)
+
+            self.assertEqual(new_x.shape, x.shape)
+            
+            if (0): #torch.cuda.is_available():
+
+                ca.to_device("cuda")
+
+                x = torch.randn(vectorization, 1, 32, 32).to("cuda")
+                new_x = ca.forward(x)
+
+                self.assertEqual(new_x.shape, x.shape)
+
+    def test_id_conv(self):
```

Comment:
`test_id_conv`, `test_id_layer` are redundant, but test_id_layer did not catch a problem when channels and groups were different. Therefore, consolidate these tests into one, and lose repetitive/redundant test assertions.